grug/types/src/events/nested.rs

```diff
@@ -57,6 +57,13 @@ impl Event {
             guest_event,
         })
     }
+
+    pub fn as_execute(self) -> EvtExecute {
+        match self {
+            Self::Execute(evt) => evt,
+            _ => panic!("not an execute event"),
+        }
+    }
```

Comment:
In other parts of the code we have a `generate_downcast!` macro for generating code like this. Can you do the same?